samples/CustomMiddleware/MyCustomMiddleware.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Middleware;
+using Microsoft.Extensions.Logging;
+
+namespace CustomMiddleware
+{
+    public class MyCustomMiddleware : IFunctionsWorkerMiddleware
+    {
+        public async Task Invoke(FunctionContext context, FunctionExecutionDelegate next)
+        {
+            // This is added pre-function execution, function will have access to this information
+            // in the context.Items dictionary
+            context.Items.Add("middlewareitem", "Hello, from middleware");
+            
+            await next(context);
```

Comment:
how can I get\override http response here? for ex. authorization, global exception handler, e.t.c ?

Reply:
@Romfos since those are more general questions about the middleware model, as opposed to this PR, it might be simpler to iterate on an issue if you can open one.

For the worker middleware, it's important to understand that the middleware pipeline in the worker is **not** HTTP centric. The context and abstractions expose information about the various triggers and bindings your functions use, one of which would be HTTP. You can use a the various properties in the context to inspect the payload and (depending on where your middleware is injected) the values for the bindings. This would be a topic that would be good to document and follow up in a separate issue.

For the exception handler, though (and other cross-cutting concerns not specific to HTTP), the pattern is the same; your middleware can handle and deal with exceptions bubbling back up after invoking the next middleware in the chain.

Reply:
@fabiocav, I wonder if the sample would be a bit more concrete it would help to understand that the middleware pipeline is a Russian doll concept, allowing to add something before Function invocation and after, that would also be able to exchange information from/to middleware and the function. For instance, measuring Function execution time. That works great before and after next()/Function invocation but doesn't demonstrate the middleware and Function interaction, passed back and forth data that is useful in a real-world scenario. Must admit, I understand what the sample is doing technically but can't think of how where it'd be useful.

Reply:
The sample tries to illustrate that flow, passing and sharing data between middleware and the function. Some more advanced features build on this to enable things like model binding in the worker. Collecting some core scenarios will be helpful so we can expand the sample, but my key goal here was to keep this simple and primarily demonstrate the registration process, as we were getting questions about this and didn't have an example we could share.

Some of the ideas for samples in this space are:
 - Cross cutting concerns
   - Error handling
   - Logging scopes
   - Authorization/payload inspection and short-circuiting
- Data enrichment
  - Additional state
  - Influence conversion logic

That's a small list, and we can expand quite a bit on this. But I think having something simple, to highlight the basic registration process, helps a lot of people get started here.

If we take the sample here, we could envision a scenario where "customer profile" information is retrieved and added to the context, we can further dive into this to understand how middleware can enrich the context with custom features (another topic that requires more documentation) to make the same scenario more efficient and drive interaction with other pieces of middleware as well (as opposed to just the function). This would also help enable scenarios like custom model binding to things a middleware is adding to the context, for example...